### PR TITLE
feat: Add primary key validation

### PR DESCRIPTION
**Pull request by yevgenypats**

### CODE REVIEW

File: schema/resource.go [hunk 1]
```diff
@@ -101,6 +101,19 @@ func (r *Resource) Columns() []string {
 	return r.Table.Columns.Names()
 }
 
+// Validates various constraints. for exmaple
+// does all primary keys has values
```
Comment (hermanschaaf) [inline schema/resource.go]:
```suggestion
// Validates various constraints. For example,
// checks that all primary keys have values.
```

File: plugins/source/scheduler_dfs.go [hunk 2]
```diff
@@ -169,6 +169,12 @@ func (p *Plugin) resolveResourcesDfs(ctx context.Context, table *schema.Table, c
 				if resolvedResource == nil {
 					return
 				}
+				tableMetrics := p.metrics.TableClient[table.Name][client.ID()]
```
Comment (hermanschaaf) [inline plugins/source/scheduler_dfs.go:172]: Maybe this lookup should happen inside the `if`, because that's the only place it's used?

**Review by hermanschaaf — APPROVED** [review]

File: schema/resource.go [hunk 3]
```diff
@@ -101,6 +101,18 @@ func (r *Resource) Columns() []string {
 	return r.Table.Columns.Names()
 }
 
+// Validates various constraints. For example: Does all primary keys has values.
```
Comment (erezrokah) [inline schema/resource.go:104]:
```suggestion
// Validates that all primary keys have values.
```

I believe the comment should reflect what the function does now. We can change the comment if/when we add more functionality

File: schema/resource.go [hunk 4]
```diff
@@ -101,6 +101,18 @@ func (r *Resource) Columns() []string {
...
+func (r *Resource) Validate() error {
+	for i, c := range r.Table.Columns {
```
Comment (erezrokah) [inline schema/resource.go:106]: How about we don't return early and report in case multiple primary keys are missing instead of only the first one?

**Review by erezrokah — APPROVED** [review]
Looks good, nothing blocking

File: plugins/source/scheduler_dfs.go (hunk 2 above)
Comment (yevgenypats) [reply plugins/source/scheduler_dfs.go]: Good catch.

File: schema/resource.go (hunk 4 above)
Comment (yevgenypats) [reply schema/resource.go:107]: Good idea. Added.

**Comment by erezrokah** [comment]
FYI, this validation caught an issue in our tests, see https://github.com/cloudquery/cloudquery/pull/6256/commits/5fc8b1f9739a0894625566767b6467f36afb5b7d and https://github.com/cloudquery/cloudquery/actions/runs/3828313454/jobs/6513763728#step:7:115
🎉

**Pull request merged**